src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/WatchImage.cs

```diff
@@ -44,22 +42,18 @@ public void CustomizeView(WatchImageCore nodeModel, NodeView nodeView)
         private void NodeModelOnPropertyChanged(object sender, PropertyChangedEventArgs args)
         {
             if (args.PropertyName != "IsUpdated") return;
-            var im = GetImageFromMirror();
-            nodeView.Dispatcher.BeginInvoke(new Action<Bitmap>(SetImageSource), new object[] { im });
+            
+            using (var im = GetImageFromMirror())
+            {
+                // The setter is invoked on the UI thread synchronously as it needs to wait for the
+                // setter to finish executing before it leaves the using scope and the Bitmap is deleted
+                nodeView.Dispatcher.Invoke(new Action<Bitmap>(SetImageSource), new object[] { im });
+            }
         }
 
-        private void SetImageSource(System.Drawing.Bitmap bmp)
+        private void SetImageSource(Bitmap bmp)
         {
-            if (bmp == null)
-                return;
-
-            // how to convert a bitmap to an imagesource http://blog.laranjee.com/how-to-convert-winforms-bitmap-to-wpf-imagesource/ 
-            // TODO - watch out for memory leaks using system.drawing.bitmaps in managed code, see here http://social.msdn.microsoft.com/Forums/en/csharpgeneral/thread/4e213af5-d546-4cc1-a8f0-462720e5fcde
-            // need to call Dispose manually somewhere, or perhaps use a WPF native structure instead of bitmap?
-
-            var hbitmap = bmp.GetHbitmap();
-            var imageSource = Imaging.CreateBitmapSourceFromHBitmap(hbitmap, IntPtr.Zero, Int32Rect.Empty, BitmapSizeOptions.FromWidthAndHeight(bmp.Width, bmp.Height));
-            image.Source = imageSource;
+            image.Source = ResourceUtilities.ConvertToImageSource(bmp);
```

Comment:
Hi @aparajit-pratap, let's try to keep that set image done asynchronously, if at all possible (we have seen twice a deadlock condition when main window is closed, the main thread tries to shut down `ISchedulerThread` while the thread is waiting for main thread to process this `Invoke` synchronous call. Here's what I'd recommend:

```csharp
private void NodeModelOnPropertyChanged(object sender, PropertyChangedEventArgs args)
{
    if (args.PropertyName != "IsUpdated")
        return;

    var data = nodeModel.CachedValue;
    var bitmap = data.Data as Bitmap;
    if (bitmap != null)
    {
        nodeView.Dispatcher.BeginInvoke(new Action<Bitmap>(SetImageSource),
        new object[] { nodeModel.CachedValue.Data as Bitmap });
    }
}

private void SetImageSource(System.Drawing.Bitmap bmp)
{
    if (bmp != null)
    {
        using (bmp)
        {
            image.Source = ResourceUtilities.ConvertToImageSource(bmp);
        }
    }
}
```

Please have a go and let's see where that leads us. Thanks!